internal/jobservice/repository/postgres.go

```diff
@@ -330,3 +316,26 @@ func (s *JSRepoPostgres) GetSubscribedJobSets(ctx context.Context) ([]Subscribed
 	}
 	return tuples, nil
 }
+
+func (s *JSRepoPostgres) purgeExpired(ctx context.Context) {
+	jobsetsStmt := fmt.Sprintf(`DELETE FROM jobsets WHERE Timestamp < (extract(epoch from now()) - %d);`, s.jobServiceConfig.PurgeJobSetTime)
+	jobServiceStmt := fmt.Sprintf(`DELETE FROM jobservice WHERE Timestamp < (extract(epoch from now()) - %d);`, s.jobServiceConfig.PurgeJobSetTime)
+
+	ticker := time.NewTicker(time.Duration(s.jobServiceConfig.PurgeJobSetTime) * time.Second)
+	log := log.WithField("JobService", "ExpiredJobSetsPurge")
+
+	for range ticker.C {
+		result, jobsetErr := s.dbpool.Exec(ctx, jobsetsStmt)
+		if jobsetErr != nil {
+			log.Error("error deleting expired jobsets: ", jobsetErr)
+		} else {
+			log.Debugf("Deleted %d expired jobsets", result.RowsAffected())
```

Comment:
I think you'll need to change this to be similar to the logic in sqlite.go. `database/sql.Result.RowsAffected()` returns a `(int 64, error)`.  (Run `go doc dtabase/sql.Result` to see details.) Right now, this code produces (on my Linux box):
```
INFO[2023-06-02T17:01:24Z] Deleted %d expired jobsets0                   JobService=ExpiredJobSetsPurge
INFO[2023-06-02T17:01:24Z] Deleted 0 expired jobs                        JobService=ExpiredJobSetsPurge
```
So if you make a separate call to RowsAffected(), and check for err not being nil, etc, it should produce better output.

Reply:
this is weird.
> I think you'll need to change this to be similar to the logic in sqlite.go. database/sql.Result.RowsAffected() returns a (int > 64, error). (Run go doc dtabase/sql.Result to see details.) Right now, this code produces (on my Linux box):

The [RowsAffected](https://pkg.go.dev/github.com/jackc/pgx/v5@v5.3.1/pgconn#CommandTag.RowsAffected) function here returns an `int64` directly, also the [Exec](https://pkg.go.dev/github.com/jackc/pgx/v5@v5.3.1/pgxpool#Conn.Exec) function here in `postgres` is different than the `sqlite` one, here it returns a `CommandTag` not `sql.Result`.

 How did you generate these logs? I will try to reproduce it and see

Reply:
I'm so embarrassed - of course you are right! I forgot that for Postgres we are using the `github.com/jackc/pgx/v5` library, which has a slightly different interface than the standard `database/sql` (which the sqlite driver complies with).  My IDE shows that the pgx library has its `Exec` also returning two values: `(CommandTag, error)`.

The way I run it is:
```
$ docker system prune                          # clean out any old containers, etc
$ mage localdev minimal testsuite        # see docs/developer/manual-localdev.md for mage install details
```

That should run successfully - it will take a few minutes. After that, I made the following changes (temporary, I did not commit them), just to exercise and see the logging:
```
diff --git a/e2e/setup/jobservice.yaml b/e2e/setup/jobservice.yaml
index a64851c2..7c82e67a 100644
--- a/e2e/setup/jobservice.yaml
+++ b/e2e/setup/jobservice.yaml
@@ -1,6 +1,6 @@
 subscribeJobSetTime: 60
 subscriberPoolSize: 30
-purgeJobSetTime: 10000
+purgeJobSetTime: 100
 databaseType: "postgres"
 postgresConfig:
   maxOpenConns: 50
diff --git a/internal/jobservice/repository/postgres.go b/internal/jobservice/repository/postgres.go
index e633b6a4..54e4d1c4 100644
--- a/internal/jobservice/repository/postgres.go
+++ b/internal/jobservice/repository/postgres.go
@@ -329,13 +329,13 @@ func (s *JSRepoPostgres) purgeExpired(ctx context.Context) {
                if jobsetErr != nil {
                        log.Error("error deleting expired jobsets: ", jobsetErr)
                } else {
-                       log.Debugf("Deleted %d expired jobsets", result.RowsAffected())
+                       log.Info("Deleted %d expired jobsets", result.RowsAffected())
                }
                result, jobServiceErr := s.dbpool.Exec(ctx, jobServiceStmt)
                if jobServiceErr != nil {
                        log.Error("error deleting expired jobs: ", jobServiceErr)
                } else {
-                       log.Debugf("Deleted %d expired jobs", result.RowsAffected())
+                       log.Infof("Deleted %d expired jobs", result.RowsAffected())
                }
        }
 }
diff --git a/third_party/airflow/tests/integration/test_airflow_operator_logic.py b/third_party/airflow/tests/integration/test_airflow_operator_logic.py
index f65ced29..e523ce57 100644
--- a/third_party/airflow/tests/integration/test_airflow_operator_logic.py
+++ b/third_party/airflow/tests/integration/test_airflow_operator_logic.py
@@ -127,7 +127,7 @@ def success_job(client: ArmadaClient, jobservice: JobServiceClient):
     assert job_message == f"Armada test:{job_id} succeeded"


-@pytest.mark.skip(reason="we should not test performance in the CI.")
+#@pytest.mark.skip(reason="we should not test performance in the CI.")
 def test_parallel_execution(client: ArmadaClient, jobservice: JobServiceClient):
     threads = []
     success_job(client=client, jobservice=jobservice)
@@ -140,7 +140,7 @@ def test_parallel_execution(client: ArmadaClient, jobservice: JobServiceClient):
         thread.join()


-@pytest.mark.skip(reason="we should not test performance in the CI.")
+# @pytest.mark.skip(reason="we should not test performance in the CI.")
 def test_parallel_execution_large(client: ArmadaClient, jobservice: JobServiceClient):
     threads = []
     success_job(client=client, jobservice=jobservice)
```
Then I ran the airflow integration tests (it's misnamed - it doesn't really run Airflow):
```
$ time make tests-e2e-airflow
```
While that is running, in a separate terminal, run:
```
$ docker logs -f jobservice
```
and watch for that logging message to show up (it may take a while).

I believe there is a way to change a running container's overall logging level to Debug from the Info default, but could not quickly find out how to do that, so I just tweaked the level in the code, to verify the message it outputs.

Reply:
After all that is done, to shut down all the containers, run
```
$ mage localdevstop
```
then run `docker ps` to verify all are shut down.

Reply:
I just ran it again now, here is a sample of the output:
```
INFO[2023-06-02T20:49:37Z] closed the event client connection
ERRO[2023-06-02T20:49:37Z] error on subscribingstream subscription ttl exceeded: 60
INFO[2023-06-02T20:49:42Z] Deleted %d expired jobsets0                   JobService=ExpiredJobSetsPurge
INFO[2023-06-02T20:49:42Z] Deleted 0 expired jobs                        JobService=ExpiredJobSetsPurge
INFO[2023-06-02T20:50:02Z] job service has 2 subscribed job sets         JobService=Startup
INFO[2023-06-02T20:50:02Z] SubscribeToJobSetId start for queue-a/test-c4c80c94-0186-11ee-927c-0242ac13000b with messageId 1685738885791:0:0:1
INFO[2023-06-02T20:50:02Z] SubscribeToJobSetId start for queue-a/test-e897d85c-0186-11ee-927c-0242ac13000b with messageId 1685738953238:0:0:0
```

Reply:
This is all on my Linux box running Ubuntu 22.04, Docker 24.0.2, all the other standard versions of our tools.

Reply:
The bug you mention here @richscott is in the diff you've posted [here](https://github.com/armadaproject/armada/pull/2499#discussion_r1214826449).
This line:
```
+                       log.Info("Deleted %d expired jobsets", result.RowsAffected())
```

Should be 
```
+                       log.Infof("Deleted %d expired jobsets", result.RowsAffected())
```

Reply:
Thanks @owenthomas17 !

Yes, as owenthomas17 mentioned, you used `log.Info` instead of `log.Infof` . A similar logs snippet to yours following the same steps and using `log.Infof`:
```shell
INFO[2023-06-03T18:31:59Z] closed the event client connection           
ERRO[2023-06-03T18:31:59Z] error on subscribingstream subscription ttl exceeded: 60 
INFO[2023-06-03T18:32:04Z] Deleted 0 expired jobsets                     JobService=ExpiredJobSetsPurge
INFO[2023-06-03T18:32:04Z] Deleted 0 expired jobs                        JobService=ExpiredJobSetsPurge

```
 I don't like logging zero deletion though, I don't think it is helpful and it will add only noises to the debug logs. I will change it to only log if `result.RowsAffected() > 0`.